bpf/tests/no_snat_endpoint_masquerade_skip_test.c

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Cilium */
+
+#include <bpf/ctx/skb.h>
+#include <bpf/api.h>
+#include "common.h"
+
+#define ENABLE_IPV4 1
+#define ENABLE_IPV6 1
+#define ENABLE_NODEPORT 1
+#define ENABLE_BPF_MASQUERADE 1
+#define ENABLE_MASQUERADE_IPV4 1
+#define ENABLE_MASQUERADE_IPV6 1
+#define IS_BPF_HOST 1
```

Comment:
I think the covered scenarios would also nicely fit as additional tests inside `bpf/tests/host_bpf_masq.c`, validating that the src IP is not re-written for such endpoints.